### ISSUE TITLE
Information request on a MO2 cosmetic issue font size and moving MO2 compatdata to save on disk space.

### ISSUE THREAD

**Issue by tetoNidan** [sonic2kk/steamtinkerlaunch#900]
## System Information
- SteamTinkerLaunch version: v12.12
- Distribution: Garuda

## Feature Description

So this request isn't so much of a feature request as it is a request for some advice. 

1. MO2's font is so small I cant read it without pressing my nose to the monitor. What would be the recommended method to increase the font size? I cant remember if you can do it in winecfg, change system wide font scale I mean? Will adjusting the font size in the winecfg STL game menu work? It is using the PFX for the game not the MO2 PFX right, so I would have to manually set the PFX with WINEPREFIX? Also if I do that mono will be installed for the winecfg GUI, will that brake anything? Asking because after I installed mono a long time ago to the MO2 PFX I lost the text to LOOT. At least that is what I think broke the text output to LOOT, really the only thing I can think of that I did to that MO2 PFX. I could be wrong about that though. It was also over a year or so ago but I was never able to fix that. I also stumbled onto this forum that said I could [use a different MO2 theme](https://stepmodifications.org/forum/topic/14088-basic-font-tiny/) but I haven't been so bold as to try it yet. Any advice is appreciated.

4. Can I just move my MO2 compatdata folder to an external drive? I currently have it installed to my $USER in it's native location. I have limited space left as I have 2 heavily modded games SKAE, FO4 and probably soon Starfield. 

I was wondering the best way to go about this. Should I move the compatdata via the Global MO2 compatdata location setting in STL? Should I reinstall MO2? Maybe I should just use a symbolic link `ln -s source dest`?

Thanks sonic2kk!

**Comment by sonic2kk** [comment]
The font scaling issue should be possible to fix using the DPI scaling option in Winecfg, under its "Graphics" tab :-)

![image](https://github.com/sonic2kk/steamtinkerlaunch/assets/7917345/76d17766-00a9-4dbc-9468-63a676fa0b1c)

ModOrganizer 2 is launched in two different ways from STL. There is Standalone Mode (where MO2 has its own prefix) and Game Mode (where it's launched in the game prefix). So if you were to change the DPI scale in winecfg (which affects the font scaling), and if you want this to apply to both ways that MO2 is used, you'd need to do it for both the MO2 prefix (from STL git you can run `steamtinkerlaunch mo2 winecfg`, not possible as far as I recall in v12.12) and for the game prefix (on STL git, you can launch Winecfg from the One-Time Run menu).

This isn't explicitly a font scaling option though, this will scale the entire DPI and will do so for every application in the prefix. In the case of Bethesda games, this will also scale parts of the game launcher. Though the game itself should be unaffected -- basically anything that looks like a "Windows UI" if you know what I mean, will be scaled. This is by design and useful (you wouldn't really want big text flowing out of a button for example) but just be careful what size you choose.

Setting values too high can make some parts of the UI inaccessible. This applies to MO2 or any other program running in a prefix with an increase value. Here's MO2 running with a DPI scale of 148, maximised on a 1080p display:

![image](https://github.com/sonic2kk/steamtinkerlaunch/assets/7917345/288e317d-0aff-4d5b-8efb-7a772ae6c389)

The lower part of the window is basically inaccessible, it is being drawn "outside" of the window borders. There will unfortunately have to be a bit of a balancing act here between getting a DPI you're comfortable with, and one that won't break UIs.

If you need to reset and forget, **the default DPI value is `96`**.

I did search around to see if there were alternatives and I couldn't find any, this seems to be the way to go.

<hr>

>  change system wide font scale I mean? Will adjusting the font size in the winecfg STL game menu work?

Changing your system font scale won't impact the Wine font size. If you're using X11 setting a different global scale should affect Wine, but the fonts themselves are managed differently, as Wine window content is (to my limited knowledge) drawn entirely independently of native applications.

You can use the Winecfg option on the STL Game Menu to launch Winecfg before MO2 starts, but this will only apply to the game prefix afaik, so if you run `steamtinkerlaunch mo2 start`, you won't have the DPI scaling, you'll have to set that manually. The easiest way to do this is with STL git and using the `steamtinkerlaunch mo2 winecfg` option, otherwise you can manually run something along these lines: `WINEPREFIX="/home/gaben/.config/steamtinkerlaunch/mo2/compatdata/pfx" WINE="/path/to/wine/executable" winecfg`

> Also if I do that mono will be installed for the winecfg GUI

This is the case for regular Wine, but it shouldn't be the case for Proton I don't think, at least testing with GE-Proton this hasn't been the case. Mono shouldn't be used with Proton at all, you shouldn't be prompted with Proton to install `wine-mono` ever. This should all be built-in and handled for you. In the [GE-Proton8-14](https://github.com/GloriousEggroll/proton-ge-custom/releases/tag/GE-Proton8-14) release notes for example that winemono is one of the internal dependencies, and this has been the case for a while. Wine-Mono is needed to display the MO2 GUI at all, any application that uses .NET (which is the vast majority of Windows native GUI programs) will depend on Wine-Mono, and MO2 should be no exception.

So you *shouldn't* get this prompt I don't think, this shouldn't need to be a concern :-)

> I also stumbled onto this forum that said I could [use a different MO2 theme](https://stepmodifications.org/forum/topic/14088-basic-font-tiny/) but I haven't been so bold as to try it yet

Yup, MO2 has a few themes (there's a NieR:Automata-themed one I particularly enjoy) but I don't know if any of them change the font size. I haven't messed around with creating a theme or modifying them either so I couldn't tell you much more at the moment. They're probably just text files (maybe XML, lots of themes use XML) and I could see a font size being something they allow you to customise internally at least.

<hr>

> Can I just move my MO2 compatdata folder to an external drive? I currently have it installed to my $USER in it's native location. I have limited space left as I have 2 heavily modded games SKAE, FO4 and probably soon Starfield.

You could link it definitely, but you could also just tell MO2 to use a different folder to store your mods. When you select a MO2 instance you can click the settings button and go to the "Paths" tab, and then change the path to the folder to store your mods (among other things). For example, my mod folder is in the MO2 prefix at `/home/user/.config/steamtinkerlaunch/mo2/compatdata/pfx/drive_c/users/steamuser/AppData/Local/ModOrganizer/Oblivion/mods` (the name before `/mods` will vary depending on your game).

I tested this a looooong time ago for similar reasons and it appeared to work, though you may have to copy mods over manually. However for Starfield you should be able to do this before you start modding, and keep SKAE and FO4 in-tact.

As a side-note: Is Starfield supported in MO2 now? If so, I should add it to `mo2games.txt`, I recently had to do it for Vortex.

<hr>

Hope this answers your questions! I'll leave this open if there's more to discuss, but feel free to close if this solves things for you. Happy gaming!

**Comment by tetoNidan** [comment]
This solves everything I needed to know! Thanks for the info and screenshots including defaults for dpi. Certainly will be helpful!

I think I will use a symbolic link as keeping MO2's default game locations sounds easier. 

> As a side-note: Is Starfield supported in MO2 now? If so, I should add it to mo2games.txt, I recently had to do it for Vortex.

Actually I don't know but I just bought the game. I have bought most my Bethesda games at a serious discount on steam sales and would like to pay the developers for the years of enjoyment they have brought me.  I do know there are already over 200+ mods for it on nexus already according to a reddit post I saw yesterday so if MO2 hasn't supported it yet I'm sure they will soon. I will give it a shot and let you know as soon as I can test. 

Downloading it now so I guess I will leave this discussion open until I can give it a go. I will get back to you soon. I don't have the fastest internet connection and I'm sure steam servers are a bit taxed to so it might be later this weekend before I get back to you.

As always, thank you very much sonic2kk!

**Comment by sonic2kk** [comment]
Starfield probably won't be detected but you should check if it's a supported game at all first. I don't see it in my instances list because I don't own it, and I don't see a plugin for it, so it's worth checking first to make sure it's something MO2 supports before flagging it as an STL issue just in case it's not supported yet :-)

I don't think there's a new MO2 release so you may have to use a MO2 beta if there is one to get access to this.

**Comment by sonic2kk** [comment]
Just checked, ModOrganizer 2 does not yet support Starfield, but there is an open PR on their repository for it: https://github.com/ModOrganizer2/modorganizer/pull/1863

This does not necessarily mean it will be merged in this state, and there is no ETA on a merge, there are also no builds available, but currently it seems you can't use Starfield and MO2.

I will add the game to `mo2games.txt` once support is merged :-)

In the meantime you should be able to use Vortex to mod Starfield (seems to work in STL git as reported in #892), or alternatively you can manually mod (which between the two options, my prefrence would manually mod before I touch Vortex).

**Comment by tetoNidan** [comment]
It looks like Starfield is not supported yet but is said to be in the works. Via this link on [nexuxmods](https://www.nexusmods.com/starfield/mods/681) 

> NOTE: The MO2 team has a working MO2 v5.0 beta on their discord server. You can use that instead, if you so desire, as it has full game support and SFSE injection. Link to their MO discord https://discord.gg/6GKR9jZ located in their Dev Builds section.

> This is just a simple plugin to allow you to run Starfield as a 'basic game'. This means it will be missing a lot of features you're used to. However, it will allow you to install mods and run the game through MO2. Once the Mod Organizer team comes out with an update, this mod will become obsolete.

I will defiantly keep you posted as I will be keeping a lookout for when it's supported. Going to see about joining that discord server so I can get more info. I'll likely just be patient and wait for the MO2 release and not use the beta version.

Just out of curiosity, when it is supported how do I go about updating MO2 to the latest version. is it as simple as 

`steamtinkelaunch mo2 download` then `steamtinkerlaunch mo2 install`? 

I guess I could head over to the STL wiki and figure it out. You have done a great job with the wiki by the way, I'd be surprised if this info is not there waiting for me to discover. To be honest I have never tried to update MO2 to it's latest versions before.

**Comment by sonic2kk** [comment]
Yeah, Starfield support will come eventually for MO2 it looks like :slightly_smiling_face: 

> Just out of curiosity, when it is supported how do I go about updating MO2 to the latest version. is it as simple as

Honestly, I am not sure :sweat_smile: I think MO2 will prompt you telling you that it needs to update itself, and it'll take care of the rest. I have never had to update MO2 either.

Though you should keep in mind that you will probably have to update SteamTinkerLaunch to the latest version for Starfield + MO2 support, in order for STL to detect Starfield as a supported MO2 game when it's added to `mo2games.txt`. This will involve updating SteamTinkerLaunch to git, which is available in multiple ways (STL-git AUR package, ProtonUp-Qt though it won't handle any optional dependencies, and gool ol' `sudo make install`).

> You have done a great job with the wiki by the way, I'd be surprised if this info is not there waiting for me to discover

I appreciate that, I've spent quite a long time writing things up on the wiki and cleaning up pages (I contributed to the wiki the most before getting deep in STL development and then eventually taking over). I will update the wiki to include this information though, as I am reasonably sure that MO2 updates itself and it'll be good to get it down for people (like me) who have only used MO2 through Wine.

I did have a realization the other day that there's no way to select a MO2 version, I believe STL always downloads the latest. Having an option to use the latest stable/beta or even specify a version just like we have now for Vortex, or even the option to just choose a custom installer executable, could be useful :-)

**Comment by sonic2kk** [comment]
[ModOrganizer 2 Wiki Page](https://github.com/sonic2kk/steamtinkerlaunch/wiki/Mod-Organizer-2#updating) was updated to note that MO2 should handle updating itself.

**Comment by tetoNidan** [comment]
> In the meantime you should be able to use Vortex to mod Starfield (seems to work in STL git as reported in https://github.com/sonic2kk/steamtinkerlaunch/issues/892), or alternatively you can manually mod (which between the two options, my prefrence would manually mod before I touch Vortex)

I've used vortex 4 or 5 years ago and it was a nightmare! I'll be patient and wait for MO2. I played vinilla oldrim and fo4 before modding, why should starfield be any different!

> I did have a realization the other day that there's no way to select a MO2 version, I believe STL always downloads the latest. Having an option to use the latest stable/beta or even specify a version just like we have now for Vortex, or even the option to just choose a custom installer executable, could be useful :-)

That might be nice if someone wants to try an older version or revert to previous version if an mo2 update has unintended issues or something like that. Honestly I havent had any issues with MO2 since STL started it's support at least no issues that I likely didn't cause my self STL just seems to work. I used to use a lutris version and it would break every update. I think your symbolic linking magic should fixed those issues though. Proton has come a long way since then too. Used to mod in vortex in a VM and move data folders around.

> [ModOrganizer 2 Wiki Page](https://github.com/sonic2kk/steamtinkerlaunch/wiki/Mod-Organizer-2#updating) was updated to note that MO2 should handle updating itself.

I will wait patiently for MO2's update and keep you posted on update results. Should I update my findings on this thread or close this and start another when things move forward?

**Comment by tetoNidan** [comment]
Also I have the STL from the chaotic-aur so I should probably update to the STL-git and make install once you have the .txt file updated huh.

**Comment by sonic2kk** [comment]
I'll close this issue for now, whenever Starfield support is in MO2 and just unavailable in STL, please feel free to open another issue telling me to get on adding support. It should be a case of adding it to `mo2games.txt` (I noticed it needs an update anyway, it's missing some games that have basic MO2 support).

Thanks! :-)

**Comment by tetoNidan** [comment]
Thanks sonic2kk I appreciate your work!

**Comment by sonic2kk** [comment]
This has become an issue for me in general with Wine, as I have recently switched to using KDE Plasma Wayland's "Apply scaling themselves" option, which fixes the issue of applications which can scale on my fractional scaling display appearing blurry, but means applications which cannot appear too small.

This has motivated me to investigate adding a Wine option for per-game, global, and eventually MO2/Vortex/HMM prefix Wine DPI options. I'm working on it right now for per-game and global, the other applications which have their own prefixes will come later.

Just wanted to give a follow-up that hopefully sometime soon this will be an option in SteamTinkerLaunch :-)

EDIT: Initial PR is up #915

**Comment by sonic2kk** [comment]
There is an (unofficial/pre-alpha?) build of ModOrganizer 2 floating around with initial Starfield support, and as such I have added Starfield to `mo2games.txt`.

This build is not tested by me, I have only heard a few minutes ago that it is available, it may not even work under Wine, and Starfield may not even be detected by MO2 with this build. Basically, I expect none of this to work, but I have added it to the games list nonetheless now that it is a bit more tangible :-)

**Comment by sonic2kk** [comment]
Link to issue which raised  this: #922
